Lib/test/test_configparser.py

```diff
@@ -980,6 +980,50 @@ def test_set_nonstring_types(self):
         self.assertRaises(TypeError, cf.set, "sect", 123, "invalid opt name!")
         self.assertRaises(TypeError, cf.add_section, 123)
 
+    def test_str(self):
+        self.maxDiff = None
+        cf = self.config_class(allow_no_value=True, delimiters=('=',), strict=True)
+        cf.add_section("sect1")
+        cf.add_section("sect2")
+        cf.add_section("sect3")
+        cf.add_section("sect4")
+        cf.add_section("sect5")
+        cf.add_section("sect6")
+        cf.set("sect1", "option1", "foo")
+        cf.set("sect2", "option2", "bar")
+
+        expected_str = (
+            "<ConfigParser: {'sect1': {'option1': 'foo'}, 'sect2': {'option2': 'bar'}, "
+            "'sect3': {}, 'sect4': {}, 'sect5': {}, 'sect6': {}}>"
+        )
+        self.assertEqual(str(cf), expected_str)
+
+    def test_repr(self):
+        self.maxDiff = None
+        cf = self.config_class(allow_no_value=True, delimiters=('=',), strict=True)
+        cf.add_section("sect1")
+        cf.add_section("sect2")
+        cf.add_section("sect3")
+        cf.add_section("sect4")
+        cf.add_section("sect5")
+        cf.add_section("sect6")
+        cf.set("sect1", "option1", "foo")
+        cf.set("sect2", "option2", "bar")
+        cf.read_string("")
+
+        dict_type = type(cf._dict).__name__
+
+        expected_repr = (
```

Comment:
This is very hard to read I think. How about doing something like this:

```py
expected = f"<{classname}({params=}, {state=})>"
```

and you create the `classname`, `params` and `state` strings yourself? (possibly with a `StringIO` or with `''.join(parts)`? as everything is inlined it's hard to read.

Reply:
okay, I have separated out these three.